packages/fxa-content-server/app/scripts/views/settings.js

```diff
@@ -142,7 +150,17 @@ const View = BaseView.extend({
   beforeRender() {
     const account = this.getSignedInAccount();
 
-    return account.fetchProfile().then(() => this.user.setAccount(account));
+    return account
+      .fetchProfile()
+      .then(() => this.user.setAccount(account))
+      .then(() => account.settingsData())
```

Comment:
Is an error message displayed if `settingsData` rejects with an error? I think it'll end up [redirecting to the 500 page with no real user information](https://github.com/mozilla/fxa/blob/c2e696500db9715eca8a42660c60c5a016335050/packages/fxa-content-server/app/scripts/views/app.js#L150) which is probably not the desired behavior. Maybe catch errors and call `this.displayError`. Of course, for that to actually print anything, you need an element with the `.error` class in the settings template where the message will be printed.

It looks like `account.fetchProfile` would have suffered from this problem for years. How have we gotten away with this for so long, or am I missing something?